app/assets/javascripts/modules/tabs.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+moj.Modules.tabs = {
+  listSelector: 'ul.tabs',
+  panelSelector: 'section.tab-panel',
+  activeClass: 'active',
+
+  init: function() {
+    var self = this;
+
+    self.$list = $(self.listSelector);
+
+    if(self.$list.length) {
+      self.$tabs = self.$list.find('li');
+      self.$panels = $(self.panelSelector);
+
+      self.storeSlugs();
+      self.showInitialTab();
+      self.bindEvents();
+    }
+  },
+
+  bindEvents: function() {
+    var self = this;
+
+    self.$list.on('click', function(e) {
+      self.selectTab($(e.target));
+    });
+  },
+
+  showInitialTab: function() {
+    var self = this,
+      docHash = document.location.hash,
+      tabIndex = 0;
+
+    if(docHash.length) {
+      var checkSlug = docHash.slice(1);
+      var slugIndex = self.slugs.indexOf(checkSlug);
+      if (slugIndex > 0) {
+        tabIndex = slugIndex;
+      }
+    }
+
+    self.showTab(tabIndex);
+  },
+
+  storeSlugs: function() {
+    var self = this;
+
+    self.slugs = [];
+    self.$tabs.each(function(x, tab) {
+      var tabSlug = encodeURIComponent($(tab).text());
```

Comment:
This seems brittle - maybe add a `data-` attribute to the tabs for the slug and use that instead of the text?

Reply:
I did it this way to keep the HTML completely clean and mean that no extra attributes were needed. Why would this be more brittle than using `data-` attributes?

Reply:
Actually, ignore me - I thought if the tab text changed it would break, but it won't

Reply:
👍